app/services/invoices/customer_usage_service.rb

```diff
@@ -60,8 +62,14 @@ def customer(customer_id: nil)
       )
     end
 
-    def subscription
-      @subscription ||= customer.active_subscription
+    def find_subscription(subscription_id)
+      return nil unless customer
+
+      active_subscriptions = customer.active_subscriptions
+
+      return nil unless active_subscriptions
+
+      active_subscriptions.find_by(id: subscription_id)
```

Comment:
Since `customer.active_subscriptions` is an ActiveQuery scope, you should do it that way:

```ruby
customer&.active_subscriptions&.find_by(id: subscription_id)
```

It will run only one DB query instead of two.

Reply:
Updated